### PR TITLE
Oop principles / it-academy.by / M-JC1-66-20 / Oxana Peregud / Done all projects

### PR DESCRIPTION
**Pull request by OxanaPeregud**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [x] approved

### CODE REVIEW

**Review by rakovets — CHANGES_REQUESTED** [review]
I used `gradle compileTestJava` and got

```
> Task :compileTestJava FAILED
```

You need to fix it.

File: src/main/java/com/rakovets/course/java/core/practice/oop/principles/battle_ground/Archer.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    private int health;
+
+    public Archer(String name, int health) {
+        super(name, health);
+        this.health = health;
+    }
+
+    @Override
+    public void attackEnemy(Enemy enemy) {
+        enemy.takeDamage(50);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/oop/principles/battle_ground/Archer.java:13]: What is `50`? Why is `50`? You need to make constant.

File: src/test/java/com/rakovets/course/java/core/practice/oop/principles/cat_home/SiameseTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                Arguments.of(35.0, new Person(50.0))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("mewProviderArguments")
+    void mew(double expectedDouble, Person person) {
+        Siamese siamese = new Siamese("Siamese");
+        siamese.mew(person);
+        assertEquals(expectedDouble, person.getHappiness());
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/oop/principles/cat_home/SiameseTest.java:30]: You could use 'Given-When-Then' approach.

**Review by rakovets — CHANGES_REQUESTED** [review]
Perfect job!
You need to fix the following problems.

File: src/test/java/com/rakovets/course/java/core/practice/oop/principles/cat_home/SiameseTest.java (hunk 2 above)
Comment (OxanaPeregud) [reply src/test/java/com/rakovets/course/java/core/practice/oop/principles/cat_home/SiameseTest.java:30]: nah.. I like it the way it is

**Review by rakovets — APPROVED** [review]

**Pull request closed without merging**